Kernel/FileSystem/ISO9660FileSystem.cpp

```diff
@@ -0,0 +1,731 @@
+                }
+            }
+
+            dbgln_if(ISO9660_VERY_DEBUG, "next(): Read directory contents");
+
+            m_current_directory.offset = 0;
+            get_header();
+            if (m_current_header->length == 0) {
+                // We have found an empty directory, let's continue with the
+                // next one.
+                if (!go_up())
+                    return false;
+            } else {
+                // We cannot skip here, as this is the first record in this
+                // extent.
+                return true;
+            }
+        }
+
+        return skip();
+    }
+
+    // Skips to the directory in the list, returns whether there was a next one.
+    // No allocation here, cannot fail.
+    bool skip()
+    {
+        VERIFY(m_current_directory.entry);
+
+        if (m_current_directory.offset >= m_current_directory.entry->length) {
+            dbgln_if(ISO9660_VERY_DEBUG, "skip(): Was at last item already");
+            return false;
+        }
+
+        m_current_directory.offset += m_current_header->length;
+        get_header();
+        if (m_current_header->length == 0) {
+            // According to ECMA 119, if a logical block contains directory
+            // records, then the leftover bytes in the logical block are
+            // all zeros. So if our directory header has a length of 0,
+            // we're probably looking at padding.
+            //
+            // Of course, this doesn't mean we're done; it only means that there
+            // are no more directory entries in *this* logical block.  If we
+            // have at least one more logical block of data length to go, we
+            // need to snap to the next logical block, because directory records
+            // cannot span multiple logical blocks.
+            u32 remaining_bytes = m_current_directory.entry->length - m_current_directory.offset;
+            if (remaining_bytes > m_fs.logical_block_size()) {
+                m_current_directory.offset += remaining_bytes % m_fs.logical_block_size();
+                get_header();
+
+                dbgln_if(ISO9660_VERY_DEBUG, "skip(): Snapped to next logical block (succeeded)");
+                return true;
+            }
+
+            dbgln_if(ISO9660_VERY_DEBUG, "skip(): Was at the last logical block, at padding now (offset {}, data length {})", m_current_directory.entry->length, m_current_directory.offset);
+            return false;
+        }
+
+        dbgln_if(ISO9660_VERY_DEBUG, "skip(): Skipped to next item");
+        return true;
+    }
+
+    bool go_up()
+    {
+        if (m_directory_stack.is_empty()) {
+            dbgln_if(ISO9660_VERY_DEBUG, "go_up(): Empty directory stack");
+            return false;
+        }
+
+        m_current_directory = m_directory_stack.take_last();
+        get_header();
+
+        dbgln_if(ISO9660_VERY_DEBUG, "go_up(): Went up a directory");
+        return true;
+    }
+
+    bool done()
```

Comment:
This sounds like a `const` operation. Should it be?

Reply:
Yes indeed. :^)